src/Containers/OhmmsPETE/OhmmsArray.h

```diff
@@ -95,6 +97,74 @@ class Array
     return X.device_data();
   }
 
+  template<typename SIZET = size_t, typename = std::is_integral<SIZET>>
+  Type_t* data(const std::array<SIZET, D>& offsets)
+  {
+    size_t offset = offsets[0];
+    for (int i = 1; i < offsets.size(); i++)
+      offset = offset * Length[i] + offsets[i];
+    return X.data() + offset;
+  }
+  template<typename SIZET = size_t, typename = std::is_integral<SIZET>>
+  const Type_t* data(const std::array<SIZET, D>& offsets) const
```

Comment:
As far as I can tell this is an std::array of indexes.
The term offset makes me think these are different.  You can have an offset into a buffer.  And in CUDA at least there is a notion of a 2D and 3D buffer, but I don't think this is that.

Reply:
put `indices` as the argument name